filament/backend/src/vulkan/VulkanPipelineCache.h

```diff
@@ -184,6 +186,26 @@ class VulkanPipelineCache : public CommandBufferObserver {
 private:
     static constexpr uint32_t ALL_COMMAND_BUFFERS = (1 << VK_MAX_COMMAND_BUFFERS) - 1;
 
+    using LayoutBundleKey = utils::bitset64; // 2 bits(vertex and fragment) per sampler * 32 samplers
+
+    static void setLayoutBundleKey(ShaderStageFlags flags, uint16_t binding, LayoutBundleKey& key);
+    static VkShaderStageFlags getShaderStageFlags(LayoutBundleKey key, uint16_t binding);
+
+    static LayoutBundleKey getLayoutBundleKey(const Program::SamplerGroupInfo& samplerGroupInfo) noexcept;
```

Comment:
I think these three helper functions can live at file scope in the cpp file, they do not need to be declared in the class.

Reply:
Ok it's done. The reason is that `LayoutBundleKey ` is a private class.